src/Datadog.Trace/Span.cs

```diff
@@ -264,6 +270,17 @@ public void Finish(DateTimeOffset finishTimestamp)
             if (shouldCloseSpan)
             {
                 Context.TraceContext.CloseSpan(this);
+                if (Log.IsEnabled(LogEventLevel.Debug))
+                {
+                    var metadata =
```

Comment:
Should we try to keep all that data in `metadata` structured instead of baking it into a single string? Like you are doing below with `"Span closed: [s_id: {0}, p_id: {1}, t_id: {2}] for {3}"`.

Reply:
I got really sick of index-based interpolation, but I guess so.
it could be useful if we change these all to named properties if we decide to use serilog for stability metrics. I don't see the benefit otherwise.

Reply:
I think we can still use string interpolation while keeping the original separate values. See [`FormattableString`](https://docs.microsoft.com/en-us/dotnet/api/system.formattablestring).

Reply:
Sorry, I don't understand. Also FormattableString isn't present until 4.6, so it would need a preprocessor directive. I can add these as indexes though, just seemed less readable to me.